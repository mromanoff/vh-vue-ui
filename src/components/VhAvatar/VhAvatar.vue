<template>
  <div
    class="VhAvatar"
    :class="computedClasses"
    v-on="$listeners"
  >
    <slot />
  </div>
</template>

<script>
export default {
  name: 'VhAvatar',
  props: {
    /**
     *  One of these colors: grey, yellow, orange, green, red, blue
     */
    color: {
      type: String,
      default: 'grey',
      validator: value => {
        return value.match(/(grey|green|yellow|orange|red|blue)/);
      },
    },
    /**
     * One of these sizes: small, medium, large
     */
    size: {
      type: String,
      default: 'medium',
      validator: value => {
        return value.match(/(small|medium|large)/);
      },
    },
  },
  computed: {
    computedClasses() {
      return {
        [`VhAvatar--${this.size}`]: this.size,
        [`VhAvatar--${this.color}`]: this.color,
      };
    },
  },
};
</script>

<style scoped>
.VhAvatar {
  align-items: center;
  border-radius: 50%;
  display: inline-flex;
  justify-content: center;
  position: relative;
  text-align: center;
  vertical-align: middle;
  background-color: var(--color--blue400);
  border-color: var(--color--blue400);
  color: var(--color--white);
  font-weight: var(--font-weight--light);
  flex: 0 0 auto;
}

/* Avatar sizes */
.VhAvatar--small {
  height: var(--VhAvatar-size--small);
  width: var(--VhAvatar-size--small);
  font-size: 12px !important;
}

.VhAvatar--medium {
  height: var(--VhAvatar-size--medium);
  width: var(--VhAvatar-size--medium);
  font-size: 16px !important;
}

.VhAvatar--large {
  height: var(--VhAvatar-size--large);
  width: var(--VhAvatar-size--large);
  font-size: 24px !important;
}

/* Avatar colors */
.VhAvatar--grey {
  background-color: var(--color--grey500);
}

.VhAvatar--yellow {
  background-color: var(--color--yellow500);
}
.VhAvatar--orange {
  background-color: var(--color--orange500);
}

.VhAvatar--green {
  background-color: var(--color--green500);
}

.VhAvatar--red {
  background-color: var(--color--red500);
}

.VhAvatar--blue {
  background-color: var(--color--blue500);
}
</style>
