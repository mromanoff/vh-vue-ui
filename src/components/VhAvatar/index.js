import VhAvatar from './VhAvatar.vue';

export default VhAvatar;
