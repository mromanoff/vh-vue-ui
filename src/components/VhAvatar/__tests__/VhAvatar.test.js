import { shallowMount } from '@vue/test-utils';
import VhAvatar from '../VhAvatar.vue';

describe('VhAvatar.vue', () => {
  let wrapper;

  it('renders default VhAvatar', () => {
    wrapper = shallowMount(VhAvatar);
    expect(wrapper.props().color).toBe('grey');
    expect(wrapper.props().size).toBe('medium');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders red avata size small', () => {
    wrapper = shallowMount(VhAvatar, {
      propsData: {
        color: 'red',
        size: 'small',
      },
    });
    expect(wrapper.props().color).toBe('red');
    expect(wrapper.props().size).toBe('small');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders black avatar size large', () => {
    wrapper = shallowMount(VhAvatar, {
      propsData: {
        color: 'blue',
        size: 'large',
      },
    });
    expect(wrapper.props().color).toBe('blue');
    expect(wrapper.props().size).toBe('large');
    expect(wrapper).toMatchSnapshot();
  });
});
