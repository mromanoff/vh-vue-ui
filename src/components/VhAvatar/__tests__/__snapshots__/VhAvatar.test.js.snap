// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VhAvatar.vue renders black avatar size large 1`] = `<div class="Avatar Avatar--large Avatar--blue"></div>`;

exports[`VhAvatar.vue renders default VhAvatar 1`] = `<div class="Avatar Avatar--medium Avatar--grey"></div>`;

exports[`VhAvatar.vue renders red avata size small 1`] = `<div class="Avatar Avatar--small Avatar--red"></div>`;
